contracts/managers/TwoAssetStrategyManager.sol

```diff
@@ -26,63 +26,88 @@ import { RebalancingLibrary } from "set-protocol-contracts/contracts/core/lib/Re
 
 import { FlexibleTimingManagerLibrary } from "./lib/FlexibleTimingManagerLibrary.sol";
 import { IAllocator } from "./allocators/IAllocator.sol";
+import { ITrigger } from "./triggers/ITrigger.sol";
 
 
 /**
  * @title TwoAssetStrategyManager
  * @author Set Protocol
  *
- * Base Rebalancing Manager contract for implementing any trading pair strategy. Allocation determinations
- * are implemented in a contract that inherits the functionality of this contract. Additionally, all allocations are
- * priced using the base contracts's Allocator contract.
+ * Rebalancing Manager contract for implementing any trading pair strategy. Allocation determinations are made
+ * base on output of Trigger contract. Max base asset allocation amount is passed in and used when bullish,
+ * allocationPrecision - bullishBaseAssetAllocation used when bearish. Additionally, all allocations are priced
+ * using the base contract's Allocator contract.
  */
 contract TwoAssetStrategyManager {
     using SafeMath for uint256;
 
     /* ============ State Variables ============ */
     ICore public coreInstance;
-    IAuctionPriceCurve public auctionLibraryInstance;
     IAllocator public allocatorInstance;
+    ITrigger public triggerInstance;
+    IAuctionPriceCurve public auctionLibraryInstance;
     IRebalancingSetToken public rebalancingSetTokenInstance;
     uint256 public baseAssetAllocation;  // Percent of base asset currently allocated in strategy
     uint256 public allocationPrecision;
+    uint256 public bullishBaseAssetAllocation;
     uint256 public auctionStartPercentage;
     uint256 public auctionEndPercentage;
     uint256 public auctionTimeToPivot;
+
+    // Time to start of confirmation period in seconds
+    uint256 public signalConfirmationMinTime;
```

Comment:
If we think we are going to apply confirmations or not to other Managers, would consider putting this state and helper functions on a separate contract that gets inherited.

The main Manager case where confirmations don't apply seems to be the RSI-like ones?